relayer/chains/bitcoin/provider.go

```diff
@@ -0,0 +1,1437 @@
+	masterPubKey, _ := hex.DecodeString(p.cfg.MasterPubKey)
+	slave1PubKey, _ := hex.DecodeString(p.cfg.Slave1PubKey)
+	slave2PubKey, _ := hex.DecodeString(p.cfg.Slave2PubKey)
+	relayersMultisigInfo := multisig.MultisigInfo{
+		PubKeys:            [][]byte{masterPubKey, slave1PubKey, slave2PubKey},
+		EcPubKeys:          nil,
+		NumberRequiredSigs: NumberRequiredSigs,
+		RecoveryPubKey:     masterPubKey,
+		RecoveryLockTime:   uint64(p.cfg.RecoveryLockTime),
+	}
+	msWallet, err := multisig.BuildMultisigWallet(&relayersMultisigInfo)
+	if err != nil {
+		p.logger.Error("failed to build multisig wallet: %v", zap.Error(err))
+		return nil, err
+	}
+
+	return msWallet, nil
+}
+
+func (p *Provider) HandleBitcoinMessageTx(message *relayTypes.Message, feeRate int, reqInputs []slaveRequestInput) ([]*multisig.Input, *multisig.MultisigWallet, *wire.MsgTx, [][]byte, int, error) {
+	msWallet, err := p.buildMultisigWallet()
+	if err != nil {
+		return nil, nil, nil, nil, 0, err
+	}
+	if feeRate == 0 {
+		feeRate, err = p.GetFastestFee()
+		if err != nil {
+			p.logger.Error("failed to get fastest fee from unisat", zap.Error(err))
+			feeRate = 1
+		}
+	}
+	inputs, msgTx, err := p.buildTxMessage(message, int64(feeRate), msWallet, reqInputs)
+	if err != nil {
+		p.logger.Error("failed to build tx message: %v", zap.Error(err))
+		return nil, nil, nil, nil, 0, err
+	}
+	relayerSigs, err := multisig.SignTapMultisig(p.cfg.RelayerPrivKey, msgTx, inputs, msWallet, 0)
+	if err != nil {
+		p.logger.Error("failed to sign tx message: %v", zap.Error(err))
+		return nil, nil, nil, nil, 0, err
+	}
+	return inputs, msWallet, msgTx, relayerSigs, feeRate, err
+}
+
+func (p *Provider) Route(ctx context.Context, message *relayTypes.Message, callback relayTypes.TxResponseFunc) error {
+	if (strings.HasSuffix(message.Src, "icon") || strings.HasSuffix(message.Src, "btc")) && strings.HasSuffix(message.Dst, "btc") {
+		p.logger.Info("starting to route message", zap.Any("message", message))
+
+		key := []byte(message.Sn.String())
+		storedRelayerMessage := StoredRelayMessage{}
+		data, _ := p.db.Get(key, nil)
+		if len(data) > 0 {
+			json.Unmarshal(data, &storedRelayerMessage)
+			if storedRelayerMessage.Status == MessageStatusSuccess {
+				p.logger.Info("Message already success", zap.Any("message", storedRelayerMessage))
+				callback(message.MessageKey(), &relayTypes.TxResponse{Code: relayTypes.Success, TxHash: storedRelayerMessage.TxHash}, nil)
+				return nil
+			}
+		}
+
+		storedMessage := StoredRelayMessage{
+			Message: message,
+			Status:  MessageStatusPending,
+		}
+		value, _ := json.Marshal(storedMessage)
+		err := p.db.Put(key, value, nil)
+		if err != nil {
+			p.logger.Error("failed to store message in LevelDB: %v", zap.Error(err))
+			return err
+		}
+		p.logger.Info("Message stored in LevelDB", zap.String("key", string(key)), zap.Any("message", message))
+
+		if p.cfg.Mode != MasterMode {
+			return nil
```

Comment:
Call the callback for cleaning the message status since this won't process this,